### ISSUE TITLE
Unable to use stage parameters for profile

### ISSUE THREAD

**Issue by fargito** [serverless/serverless#10642]
### Are you certain it's a bug?

- [X] Yes, it looks like a bug

### Is the issue caused by a plugin?

- [X] It is not a plugin issue

### Are you using the latest version?

- [X] Yes, I'm using the latest version

### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Issue description

Hello ! I was trying to implement the stage parameters feature from the V3 and unfortunately it seems that the `provider.profile` property cannot use this feature, is it a bug or is it just not supported?

Thanks a lot for your help.

### Service configuration (serverless.yml) content

```yaml
const serverlessConfiguration: AWS & ServerlessContracts = {
  service: `${projectName}-core`, // Keep it short to have role name below 64
  frameworkVersion: '>=3.0.0',
  configValidationMode: 'error',
  plugins: [],
  provider: {
    name: 'aws',
    runtime: 'nodejs14.x',
    architecture: 'arm64',
    region: 'eu-west-1',
    profile: '${param:profile}', // Used to point to the right AWS account
    stage: "${opt:stage, 'dev'}",
    environment: {
      AWS_NODEJS_CONNECTION_REUSE_ENABLED: '1',
      NODE_OPTIONS: '--enable-source-maps --stack-trace-limit=1000',
    },
    eventBridge: {
      useCloudFormation: true,
    },
    httpApi: {
      payload: '2.0',
      cors: {
        // @ts-ignore we use a configuration per environment so we put it as a serverless variable
        allowedOrigins: '${param:apiGatewayCorsAllowedOrigins}',
        allowedHeaders: ['Content-Type', 'Authorization', 'Origin'],
        allowedMethods: ['GET', 'POST', 'PUT', 'PATCH', 'DELETE', 'OPTIONS'],
        allowCredentials: true,
      },
      metrics: true,
    },
  },
  functions,
  package: { individually: true },
  params: {
    dev: {
      profile: 'sls-monorepo-developer',
      apiGatewayCorsAllowedOrigins: ['http://localhost:3000'],
    },
    staging: {
      profile: '',
      apiGatewayCorsAllowedOrigins: ['https://staging.my-domain.com'],
    },
    production: {
      profile: '',
      apiGatewayCorsAllowedOrigins: ['https://www.my-domain.com'],
    },
  },
  custom: {
    projectName,
    esbuild: sharedEsbuildConfig,
  },
  contracts: {
    provides: {
      httpApiResourceContract: httpApiResourceContract.fullContractSchema,
    },
    consumes: {},
  },
  resources: {
    Description: 'Core service',
    Outputs: {
      HttpApiId: httpApiResourceContract.exportValue({
        description: 'The shared httpApi resource',
        value: { Ref: 'HttpApi' },
      }),
    },
  },
};
```

### Command name and used flags

serverless deploy

### Command output

```shell
Environment: linux, node 14.18.2, framework 3.1.1 (local), plugin 6.0.0, SDK 4.3.1
Docs:        docs.serverless.com
Support:     forum.serverless.com
Bugs:        github.com/serverless/serverless/issues

Error:
Cannot resolve serverless.ts: "provider.profile" property is not accessible (configured behind variables which cannot be resolved at this stage)
```

### Environment information

```shell
Framework Core: 3.1.1 (local)
Plugin: 6.0.0
SDK: 4.3.1
```

**Comment by mnapoli** [comment]
I'll let Piotr or Mariusz confirm next week but AFAIR this message indeed means we can't use variables for the property:

> configured behind variables which cannot be resolved at this stage

If that's indeed the case, we should maybe reword that message because I think we had a similar question for the `service` property. Maybe something like:

> "provider.profile" property is not accessible: variables cannot be used in this special property.

Anyway that's just an idea, let's wait for confirmation.

**Comment by fargito** [comment]
Well the message would not be explicit either since variables are in fact usable for `profile`, just not stage parameters

**Comment by pgrzesik** [comment]
Hello 👋 I will dive deeper, but since the `params` can also be resolved with dashboard, it seems like the resolution of it is influenced by the resolution of `region`, which in turn depends on `profile`. However, if dashboard is not used, we should still be able to resolve it without issues. I will dive into the code and report back with findings.

**Comment by medikoo** [comment]
Issues is caused by subtle shortcomings in variable resolution queuing. I've fixed that in https://github.com/serverless/serverless/pull/10652

**Comment by fargito** [comment]
Thanks a lot :rocket:

**Comment by fargito** [comment]
I can confirm that this issue is fixed with version 3.2.0
